### PR TITLE
Fix the spelling of "DigitalOcean"

### PR DESCRIPTION
**Pull request by mamercad**
## Proposed Commit Message

Fix the spelling of "DigitalOcean"

```
The name "DigitalOcean" doesn't have a space in it; it's a single compound word written in Pascal case (upper camel case).
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [ ] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: doc/rtd/topics/datasources/digitalocean.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .. _datasource_digital_ocean:
 
-Digital Ocean
+DigitalOcean
 =============
```
Comment (OddBloke) [inline doc/rtd/topics/datasources/digitalocean.rst:4]:
```suggestion
DigitalOcean
============
```

**Review by OddBloke — APPROVED** [review]
+1 (with a minor formatting fix that I'll apply), need confirmation from Canonical folks that the CLA signature is present and correct.

**Comment by TheRealFalcon** [comment]
@OddBloke yep! Already confirmed the CLA. Thanks!

**Comment by OddBloke** [comment]
Thanks @mamercad!

**Pull request merged**